atdgen/src/validate.ml

```diff
@@ -22,12 +22,34 @@ type validate_repr = (string option * bool)
                                      in addition to scanning sub-fields
     *)
 
+
+(* Matches strings like ["\""], ["\\\""], and ["\\\\\""] *)
+let escaped_quote = Str.regexp "\\(\\\\\\)*\""
+
+(* Increments all escaped quotation levels in a string
+   E.g., turns ["\"dog\" has 3 letters"] into ["\\\"dog\\\" has 3 letters"]
+*)
+let increase_escaped_quote_levels s =
+  let increase_escaped_quote_level = function
+    | Str.Text s  -> s
+    | Str.Delim s ->
+        let escapes_needed = String.length s in
+        let new_escapes = String.init escapes_needed (fun _ -> '\\') in
+        (new_escapes ^ s)
+  in
+  Str.full_split escaped_quote s
+  |> List.map increase_escaped_quote_level
+  |> String.concat ""
```

Comment:
oh you didn't have to go through that trouble. OCaml has this built-in: `String.escaped` for escaping string content without the enclosing double-quotes (`ab"c` -> `ab\"c`) or `sprintf "%S"` for creating a full quoted string literal (`ab"c` -> `"ab\"c").

Reply:
Oh doh! My initial searches didn't turn up anything, but apparently I didn't look well enough. Thanks!